napari/layers/points/points.py

```diff
@@ -175,6 +254,13 @@ def __init__(
         self._data = np.asarray(data)
         self.dims.clip = False
 
+        # Save the properties
+        if properties is None:
+            properties = {}
+        elif not isinstance(properties, dict):
+            properties = dataframe_to_properties(properties)
```

Comment:
So, I'm not positive that *everything* will work, but most things should, if `dataframe_to_properties` is just `dict`. Then you get a dict of keys -> series, and boolean indexing still works fine:

```python
In [1]: df = pd.DataFrame({'hello': [1, 2], 'goodbye': [3, 4]})
In [2]: dict(df)
Out[2]:
{'hello': 0    1
 1    2
 Name: hello, dtype: int64, 'goodbye': 0    3
 1    4
 Name: goodbye, dtype: int64}

In [3]: d = dict(df)
In [4]: d['hello']
Out[4]:
0    1
1    2
Name: hello, dtype: int64

In [5]: type(d['hello'])
Out[5]: pandas.core.series.Series

In [6]: x = d['hello']
In [7]: x[[True, False]]
Out[7]:
0    1
Name: hello, dtype: int64
```